packages/driver/src/cy/stability.ts

```diff
@@ -54,10 +54,20 @@ export const create = (Cypress, state) => ({
   },
 
   whenStableOrAnticipatingCrossOriginResponse (fn, command) {
-    const commandIsOrigin = command?.get('name') === 'origin' || false
-
-    // origin is a special command that can continue even when unstable.
-    if ((!!state('anticipatingCrossOriginResponse') && commandIsOrigin) || state('isStable') !== false) {
+    const commandIsOrigin = command?.get('name') === 'origin'
+    const commandIsEndLogGroup = command?.get('name') === 'end-logGroup'
+
+    if (
+      // cy.origin() needs to run when unstable (if we're anticipating
+      // a cross-origin response) in order to allow it to set up a spec bridge
+      // before the page loads and stability is restored
+      (!!state('anticipatingCrossOriginResponse') && commandIsOrigin)
+      // the end-logGroup command is inserted internally to mark the end of
+      // a command group and needs to be allowed or stability will hang things
+      // up if chaining cy.origin commands
+      || commandIsEndLogGroup
```

Comment:
Does it make sense here to add a test that makes sure the last command invoked in `cy.origin()` is the `end-logGroup` or is that more so testing the implementation details of command grouping?

Reply:
Added assertion in https://github.com/cypress-io/cypress/pull/21046/commits/ed121458004f01c095b8aef2f575634cc0f7a917